### ISSUE TITLE
Typescript examples that use an async invoke don't unwrap correctly when looped

### ISSUE THREAD

**Issue by leezen** [pulumi/pulumi#12507]
Example: https://www.pulumi.com/docs/reference/pkg/aws/ec2/getroutetables/

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const rts = pulumi.output(aws.ec2.getRouteTables({
    filters: [{
        name: "tag:kubernetes.io/kops/role",
        values: ["private*"],
    }],
    vpcId: var_vpc_id,
}, { async: true }));
const route: aws.ec2.Route[] = [];
for (let i = 0; i < rts.apply(rts => rts.ids.length); i++) {
    route.push(new aws.ec2.Route(`r-${i}`, {
        destinationCidrBlock: "10.0.1.0/22",
        routeTableId: rts.apply(rts => rts.ids[i]),
        vpcPeeringConnectionId: "pcx-0e9a7a9ecd137dc54",
    }));
}
```

**Comment by Zaid-Ajaj** [comment]
I think this is fixed as of #12749 but I can't find the original example program to repro and confirm the fix

**Comment by Zaid-Ajaj** [comment]
I've managed to come up with a repro example in terraform code:
```tf
variable localGatewayVirtualInterfaceGroupId {
  type = string
}

data "aws_ec2_local_gateway_route_tables" "rts" {
    filter {
        name   = "tag:kubernetes.io/kops/role"
        values = ["private*"]
    }
}

resource "aws_ec2_local_gateway_route" "routes" {
  count = length(data.aws_ec2_local_gateway_route_tables.rts.ids)
  destination_cidr_block  = "10.0.1.0/22"
  local_gateway_route_table_id = data.aws_ec2_local_gateway_route_tables.rts.ids[count.index]
  local_gateway_virtual_interface_group_id = var.localGatewayVirtualInterfaceGroupId
}
```
When converted to TypeScript using Pulumi CLI v3.76.1 and pulumi-converter-terraform v1.0.4 it generates:
```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const config = new pulumi.Config();
const localGatewayVirtualInterfaceGroupId = config.require("localGatewayVirtualInterfaceGroupId");
const rts = aws.ec2.getLocalGatewayRouteTablesOutput({
    filters: [{
        name: "tag:kubernetes.io/kops/role",
        values: ["private*"],
    }],
});
const routes: aws.ec2.LocalGatewayRoute[] = [];
rts.apply(rts => rts.ids).length.apply(rangeBody => {
    for (const range = {value: 0}; range.value < rangeBody; range.value++) {
        routes.push(new aws.ec2.LocalGatewayRoute(`routes-${range.value}`, {
            destinationCidrBlock: "10.0.1.0/22",
            localGatewayRouteTableId: rts,
            localGatewayVirtualInterfaceGroupId: localGatewayVirtualInterfaceGroupId,
        }));
    }
});

```
So the unwrapping of output-versioned invokes is working but there is an unrelated bug:
```diff
- localGatewayRouteTableId: rts,
+ localGatewayRouteTableId: rts.ids[range.value],
```
Will look into it
